apps/design/frontend/src/export_screen.tsx

```diff
@@ -1,22 +1,58 @@
-import { H1, P, Button, MainContent, MainHeader } from '@votingworks/ui';
+import { useState } from 'react';
+import {
+  H1,
+  P,
+  Button,
+  MainContent,
+  MainHeader,
+  useQueryChangeListener,
+  LoadingButton,
+  Icons,
+} from '@votingworks/ui';
 import fileDownload from 'js-file-download';
 import { useParams } from 'react-router-dom';
 import { getDisplayElectionHash } from '@votingworks/types';
+import { assertDefined } from '@votingworks/basics';
 import {
   exportAllBallots,
   exportElectionPackage,
   exportTestDecks,
+  getElectionPackage,
 } from './api';
 import { ElectionNavScreen } from './nav_screen';
 import { ElectionIdParams } from './routes';
+import { downloadFile } from './utils';
 
-export function ExportScreen(): JSX.Element {
+export function ExportScreen(): JSX.Element | null {
   const { electionId } = useParams<ElectionIdParams>();
+
+  const electionPackageQuery = getElectionPackage.useQuery(electionId);
   const exportAllBallotsMutation = exportAllBallots.useMutation();
-  const exportTestDecksMutation = exportTestDecks.useMutation();
   const exportElectionPackageMutation = exportElectionPackage.useMutation();
+  const exportTestDecksMutation = exportTestDecks.useMutation();
+
+  const [exportError, setExportError] = useState<string | undefined>(undefined);
+
+  useQueryChangeListener(electionPackageQuery, {
+    onChange: (currentElectionPackage, previousElectionPackage) => {
+      const taskJustCompleted = Boolean(
+        previousElectionPackage?.task &&
+          !previousElectionPackage.task.completedAt &&
+          currentElectionPackage.task?.completedAt
+      );
+      if (taskJustCompleted) {
+        const { error } = assertDefined(currentElectionPackage.task);
+        if (error) {
+          setExportError(error);
+        } else {
+          downloadFile(assertDefined(currentElectionPackage.url));
```

Comment:
As noted in the PR description, we may want to distinguish the act of generating the election package and downloading it, down the road.

Right now, after the background task finishes, we auto-download the file. If you miss the background task's transition from being in progress to being finished, e.g., because you close your browser and reopen it, you'll just have to click the export button again.

The backend could be smart and take advantage of work from past runs so that election package generation doesn't do everything from scratch every time, unless it needs to because the election definition has changed. This would give us all the benefits of separate generate and download buttons, without the potential confusion.